csp_billing_adapter_microsoft/plugin.py

```diff
@@ -71,6 +70,102 @@ def meter_billing(
+                'quantity': quantity,
+                'dimension': dimension_name,
+                'effectiveStartTime': str(timestamp.astimezone(timezone.utc)),
+                'planId': os.environ['PLAN_ID']
+            }
+        )
+
+    if metered_billing_data["request"]:
+        # Setup request header for the usage event API
+        metered_billing_header = {
+            'Content-type': 'application/json',
+            'x-ms-correlationid': str(uuid.uuid4()),
+            'authorization': _get_msi_token()
+        }
+
+        # Make request to the usage event API
+        data_request = urllib.request.Request(
+            'https://marketplaceapi.microsoft.com/api/batchUsageEvent'
+            '?api-version=2018-08-31',
+            data=json.dumps(metered_billing_data).encode(),
+            headers=metered_billing_header,
+            method='POST'
+        )
+
+        retries = 3
+        exc = None
+        response = None
+
+        while retries > 0:
+            try:
+                with urllib.request.urlopen(data_request) as url_open_return:
+                    response = json.loads(
+                        url_open_return.read().decode("utf-8")
+                    )
+                    break
+            except urllib.error.URLError as error:
+                exc = error
+                retries -= 1
+                continue
+
+        if exc:
+            msg = (
+                f"Failed to meter bill dimensions "
+                f"{dimensions}: {str(exc)}"
+            )
+            for dimension_name in dimensions:
+                status[dimension_name] = {
+                    "status": "failed",
+                    "error": msg
+                }
+            log.error(msg)
+            return status
+
+        if response and (response.get("count", 0) > 0):
+            for resp in response["result"]:
+                if resp["status"] == "Accepted":
+                    record_id = resp.get("usageEventId", None)
+                    log.info(
+                        'New metered billing record added with ID %s:',
+                        record_id
+                    )
+                    status[resp["dimension"]] = {
+                        "record_id": record_id,
+                        "status": "submitted"
+                    }
+                else:
+                    log.error(
+                        'Unable to log metered billing record: %s',
+                        resp
+                    )
+                    status[resp["dimension"]] = {
+                        "status": "failed",
+                        "error":
+                            f'Failed to meter bill dimensions: '
+                            f'Status: {resp["status"]} '
+                            f'Message: {resp["error"]["message"]}'
+                    }
+        return status
+
+    log.info('Nothing to meter bill: No tiers have non zero quantity values')
```

Comment:
Probably should be dimensions rather than tiers
```suggestion
    log.info('Nothing to meter bill: No dimensions have non zero quantity values')
```

Reply:
fixed

Reply:
Yes, `tiers` do not apply to SUMa for example where 1 dimension is equivalent to a product.